packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -941,104 +936,30 @@ class HotRunner extends ResidentRunner {
   static const String _pubCachePathWindows = 'Pub/Cache';
 
-  Map<String, Uri> _packageMap;
-  final String _packagesPath;
-  final FlutterProject _flutterProject;
-  final Map<String, int> _updateTime = <String, int>{};
-  int _packagesUpdateTime;
-
-  Map<String, int> get updateTime => _updateTime;
-
-  @visibleForTesting
-  Map<String, Uri> get packageMap => _packageMap;
-
-  static void _computePackageMap(Map<String, Uri> packageMap, FlutterProject flutterProject) {
-    if (flutterProject != null && flutterProject.pubspecFile.existsSync()) {
-      try {
-        final YamlMap pubspec = loadYamlDocument(flutterProject.pubspecFile.readAsStringSync()).contents;
-        final YamlMap dependencies = pubspec['dependencies'];
-        final Set<String> relevantDependencies = Set<String>.from(dependencies.keys);
-        // Remove any packages which were tagged as dev dependenices,
-        // But don't remove the app itself!
-        for (String packageName in packageMap.keys.toList()) {
-          if (!relevantDependencies.contains(packageName) && packageName != flutterProject.manifest.appName) {
-            packageMap.remove(packageName);
-            continue;
-          }
-        }
-      } catch (err) {
-        // If we detect a pubspec formatting problem, fallback to the packages file.
-      }
-    }
-    // Remove any packages which are derived from the pub cache.
-    for (String packageName in packageMap.keys.toList()) {
-      final String path = packageMap[packageName].path;
-      if ((platform.isWindows && path.contains(_pubCachePathWindows))
-          || path.contains(_pubCachePathLinuxAndWindows)) {
-        packageMap.remove(packageName);
+  static List<Uri> findInvalidated({@required DateTime lastCompiled,
+    @required List<Uri> urisToMonitor}) {
+    final List<Uri> invalidatedFiles = <Uri>[];
+    int scanned = 0;
+    final Stopwatch stopwatch = Stopwatch()..start();
+    for (Uri uri in urisToMonitor) {
+      if ((platform.isWindows && uri.path.contains(_pubCachePathWindows))
+          || uri.path.contains(_pubCachePathLinuxAndWindows)) {
+        // Don't watch pub cache directories to speed things up a little.
+        continue;
       }
-    }
-  }
-
-  List<String> findInvalidated() {
-    final File packagesFile = fs.file(_packagesPath);
-    if (packagesFile.existsSync()) {
-      final int newPackagesUpdateTime = packagesFile.statSync().modified.millisecondsSinceEpoch;
-      // Hot reloading with an updated package will often times kill a non-trivial
-      // appliction. This _might_ work, given certain application size and package
-      // constraints, so instead of exiting we print a warning so that the user has
-      // some hint on what went wrong.
-      if (newPackagesUpdateTime > _packagesUpdateTime) {
-        printError('Warning: updated dependencies detected. The Flutter application will require a restart to safely use new packages.');
+      final int updatedAt = fs
```

Comment:
Since we're only checking the files the frontend server used in the last compilation, it is possible that one or more files were deleted since then right?

In this case, I believe statSync will return null

Reply:
Oh, right, thanks! Fixed.